sdk/include/xdk/amd64/ke.h

```diff
@@ -320,11 +329,20 @@ KeGetPcr(VOID)
     return (PKPCR)__readgsqword(FIELD_OFFSET(KPCR, Self));
 }
 
+#if (NTDDI_VERSION >= NTDDI_WIN7)
+_CRT_DEPRECATE_TEXT("KeGetCurrentProcessorNumber is deprecated. Use KeGetCurrentProcessorNumberEx or KeGetCurrentProcessorIndex instead.")
+#endif
 FORCEINLINE
 ULONG
 KeGetCurrentProcessorNumber(VOID)
 {
-    return (ULONG)__readgsword(0x184);
+#if (NTDDI_VERSION >= NTDDI_VISTA)
+    return __readgsword(0x184);
```

Comment:
What's this based on? Can't find anything similar (or a matching KPCR field) in any WDK version I have.

Reply:
It's KPRCB.Number. which is USHORT on Vista and UCHAR on 2k3. I couldn't find Vista WDK, but on the WDKs I checked (8+) there was only __readgsbyte(0x184). I guess it doesn't make a difference, since Vista supports a maximum of 64 processors anyway. I'll remove the uselesss parts.

Reply:
If we want to be better and support more than 64 processors what should we do there?

Reply:
Rewrite the kernel to be more like Windows 10?